### PR TITLE
Implement Path::try_exists

### PR DESCRIPTION
**Pull request by schneems**
Implements the `Path::try_exists` associated function mentioned in #46. The aliased function was stabilized in 1.63 so to preserve compatibility with older Rust versions (ci targets 1.40) a new feature flag was added.

### CODE REVIEW

File: Cargo.toml
Comment (schneems) [inline Cargo.toml:23]: I'm not sure best practice on features. Other things I considered include:

Adding an `all` feature that loads all features:


```
all = ["path_try_exists", "io_safety"]
```

Alternatively making a feature named `1_63` that would enable all features on 1.63+.

```
1_63 = ["path_try_exists", "io_safety"]
```

File: src/errors.rs
Comment (schneems) [inline src/errors.rs:26]: In order to put the method behind a feature flag, I needed to get clippy to not complain about this unused variant. Maybe there's a better way or pattern.

File: src/path.rs
Comment (schneems) [inline src/path.rs:1]: This import is only needed when `path_try_exists` feature is enabled. Without the allow unused imports clippy generates a warning. Maybe there's a better way to avoid it. 

In this case I think it affects the whole file, not just this one import. Which is not ideal

File: src/path.rs
Comment (schneems) [inline src/path.rs:32]: Weirdly enough, `try_exists` on Path is stabilized by `fs::try_exists` is not. I couldn't follow the same pattern used by the rest of the associated trait functions.

**Comment by LPGhatguy** [comment]
Instead of a feature flag, what do you think about using one of the available crates that does feature or version detection of the running compiler?

I'm not sure what the other maintainers of fs-err would prefer, but I know that [autocfg](https://crates.io/crates/autocfg/) is a solution that would make this API available for new Rust versions without breaking older versions.

**Comment by andrewhickman** [comment]
I think using autocfg is a good idea too :+1:

**Comment by schneems** [comment]
I updated to use `autocfg` though I'm new to that library/approach so I might have missed something fundamental. I need approval for a new CI run, I've not actually tried it on an older rustc version. It compiles fine on my local machine.

I didn't remove the existing feature flag for `io_safety`, I think that would need a major version rev. This is scoped only to add support for this one function.

**Review by andrewhickman — APPROVED** [review]

**Comment by andrewhickman** [comment]
Thanks! I'll publish version 2.10.0 with this change

**Comment by schneems** [comment]
Thanks a ton for the review and for this library!

**Pull request merged**
